worker-taskstowing/src/main/java/com/microfocus/caf/worker/taskstowing/factory/TaskStowingWorkerFactory.java

```diff
@@ -0,0 +1,155 @@
+import com.microfocus.caf.worker.taskstowing.database.DatabaseClient;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class TaskStowingWorkerFactory implements WorkerFactory, NotIntendedTaskMessageForwardingEvaluator, BulkWorker
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(TaskStowingWorkerFactory.class);
+
+    private final TaskStowingWorkerConfiguration configuration;
+    private final Codec codec;
+    private final DatabaseClient databaseClient;
+
+    public TaskStowingWorkerFactory(final ConfigurationSource configSource, final DataStore dataStore, final Codec codec)
+        throws WorkerException
+    {
+        try {
+            this.configuration = configSource.getConfiguration(TaskStowingWorkerConfiguration.class);
+        } catch (ConfigurationException e) {
+            throw new WorkerException("Failed to load Task Stowing Worker configuration", e);
+        }
+        this.codec = Objects.requireNonNull(codec);
+        this.databaseClient = new DatabaseClient(configuration);
+    }
+
+    @Override
+    public Worker getWorker(WorkerTaskData workerTask) throws TaskRejectedException, InvalidTaskException
+    {
+        return new TaskStowingWorker(workerTask, configuration.getOutputQueue(), codec, databaseClient);
+    }
+
+    @Override
+    public String getInvalidTaskQueue()
+    {
+        return configuration.getFailureQueue();
+    }
+
+    @Override
+    public int getWorkerThreads()
+    {
+        return configuration.getThreads();
+    }
+
+    @Override
+    public HealthResult healthCheck()
+    {
+        try {
+            databaseClient.checkHealth();
+            return HealthResult.RESULT_HEALTHY;
+        } catch (final Exception exception) {
+            final String message = "Database is unhealthy.";
+            LOGGER.error(message, exception);
+            return new HealthResult(HealthStatus.UNHEALTHY, message);
+        }
+    }
+
+    @Override
+    public TaskForwardingAction determineForwardingAction(
+        final TaskMessage tm,
+        final TaskInformation taskInformation,
+        final boolean poison,
+        final Map<String, Object> headers,
+        final Codec codec,
+        final JobStatus jobStatus,
+        final WorkerCallback callback)
+    {
+        return jobStatus == JobStatus.Paused
+            ? TaskForwardingAction.Execute
+            : TaskForwardingAction.Forward;
+    }
+
+    @Override
+    public void processTasks(final BulkWorkerRuntime bulkWorkerRuntime) throws InterruptedException
+    {
+        final long maxBatchTime = configuration.getMaxBatchTime();
+        final int maxBatchSize = configuration.getMaxBatchSize();
+        final long cutoffTime = System.currentTimeMillis() + maxBatchTime;
+        final List<WorkerTask> workerTasks = new ArrayList<>();
```

Comment:
Rather than holding the list of worker tasks what you really should do here is create an object which holds in its state the  `partitionIdList`, `jobIdList`, `taskClassifierList`, etc.  Then where you are calling `workerTasks.add()` below you should call into this object passing the `workerTask` as a parameter.  At that point you can do a lot of what you are doing in the first half of `processTasks()` - validate the worker task and either set a response on it if it is faulty or populate the parallel lists.  Doing it like this means that you get the work of populating the parallel lists done while you are waiting for more messages to arrive.  When the cutoff time comes all you will already have the structures prepared so that you can immediately call the database.